src/I18n/DateFormatTrait.php

```diff
@@ -286,11 +293,13 @@ public static function setToStringFormat($format)
      *
      * @see \Cake\I18n\Time::i18nFormat()
      * @param string|array|int $format Format.
+     * @param bool $useI18nFormat When set to false, using `format` method when converting this object to json.
      * @return void
      */
-    public static function setJsonEncodeFormat($format)
+    public static function setJsonEncodeFormat($format, $useI18nFormat = true)
```

Comment:
I think adding a parameter here and defaulting to true will be hard to explain in the docblock.  Unfortunately, this function does not include i18n in the name like i18nFormat() does.

However, if you want a non-i18n serialization for speed, maybe a separate function to enable/disable might work.

Reply:
Switching between 2 types of placeholders is also a bad internal implementation.

Reply:
> I think adding a parameter here and defaulting to true will be hard to explain in the docblock.

I think so too.
How about adding a method `setJsonEncodeFormatAsNoneI18n($format)` ?